lib/ui/painting/canvas.cc

```diff
@@ -293,10 +293,11 @@ void Canvas::drawImage(const CanvasImage* image,
   if (!image)
     Dart_ThrowException(
         ToDart("Canvas.drawImage called with non-genuine Image."));
+  image_allocation_size_ += image->GetAllocationSize();
   canvas_->drawImage(image->image(), x, y, paint.paint());
```

Comment:
Is this accurate?  I would have expected the canvas to rasterize the drawn image at some point so it doesn't actually hold on to the full image.

Reply:
It's going to hold a reference to the SkImage as long as its alive. The memory is not necessarily duplicated (e.g. we don't necessarily have 2x the pixel allocation for the extra reference), but we need to do a better job of letting the GC know how expensive the object is, otherwise it will think it can skip a GC pass when it really needs one.

Reply:
It might help to clarify the semantics of "allocation size". The allocation size is not the size of the allocation that will be collected when the owning object is collected. Instead, it is the size of the allocation that is guaranteed to stay alive pending the collection of the owning object.

Reply:
As in a doc comment in dart_wrappable.h?

Reply:
Sure. I don't know of a better place.